src/ds/dllist.h

```diff
@@ -6,16 +6,24 @@
 
 namespace snmalloc
 {
-  template<class T, uintptr_t terminator = 0>
+  template<class T, uintptr_t terminator_ = 0>
```

Comment:
Can we make this a type with a default constructor, rather than uintptr_t?  That would then let us use `nullptr_t`, which is default constructed as `nullptr`.  If we require the type to be implicitly convertable to `T*` (which `nullptr_t` is already) then this also removes the need for the `terminator()` function.  The following is fine:

```c++
T *a = std::nullptr_t();
```

We can avoid depending on `<cstddef>` by using `decltype(nullptr)` as the default, rather than `nullptr_t`.